src/main/train.cpp

```diff
@@ -591,38 +640,15 @@ int main(int argc, char *argv[]) {
         }
     }
     string dataset = toLower(config.dataset);
-    if(dataset != "") {
-        if(dataset == "mnist") {
-            config.dataDir = "../data/mnist";
-            config.trainFile = "train-images-idx3-ubyte";
-            config.validateFile = "t10k-images-idx3-ubyte";
-        } else if(dataset == "norb") {
-            config.dataDir = "../data/norb";
-            config.trainFile = "training-shuffled-dat.mat";
-            config.validateFile = "testing-sampled-dat.mat";
-        } else if(dataset == "cifar10") {
-            config.dataDir = "../data/cifar10";
-            config.trainFile = "train-dat.mat";
-            config.validateFile = "test-dat.mat";
-        } else if(dataset == "kgsgo") {
-            config.dataDir = "../data/kgsgo";
-            config.trainFile = "kgsgo-train10k-v2.dat";
-            config.validateFile = "kgsgo-test-v2.dat";
-            config.loadOnDemand = 1;
-        } else if(dataset == "kgsgoall") {
-            config.dataDir = "../data/kgsgo";
-            config.trainFile = "kgsgo-trainall-v2.dat";
-            config.validateFile = "kgsgo-test-v2.dat";
-            config.loadOnDemand = 1;
-        } else {
-            cout << "dataset " << dataset << " not known.  please choose from: mnist, norb, cifar10, kgsgo" << endl;
-            return -1;
-        }
-        cout << "Using dataset " << dataset << ":" << endl;
-        cout << "   datadir: " << config.dataDir << ":" << endl;
-        cout << "   trainfile: " << config.trainFile << ":" << endl;
-        cout << "   validatefile: " << config.validateFile << ":" << endl;
-    }
+
+    if(setDefaults(config, dataset) == -1)
+      return -1;
+
+    cout << "Using dataset " << dataset << ":" << endl;
```

Comment:
You've made it so that it will always print this info, even if a dataset isn't used.